include/zephyr/net/net_event.h

```diff
@@ -441,6 +451,34 @@ struct net_event_ipv6_pe_filter {
 	bool is_deny_list;
 };
 
+/**
+ * @brief Network Management event information structure
+ * Used to pass information on network event
+ *   NET_EVENT_IPV4_PMTU_CHANGED
+ * when CONFIG_NET_MGMT_EVENT_INFO enabled and event generator pass the
+ * information.
+ */
+struct net_event_ipv4_pmtu_info {
+	/** IPv4 address */
+	struct in_addr dst;
+	/** New MTU */
+	uint16_t mtu;
+};
+
+/**
+ * @brief Network Management event information structure
+ * Used to pass information on network event
+ *   NET_EVENT_IPV6_PMTU_CHANGED
+ * when CONFIG_NET_MGMT_EVENT_INFO enabled and event generator pass the
+ * information.
+ */
+struct net_event_ipv6_pmtu_info {
+	/** IPv6 address */
+	struct in6_addr dst;
+	/** New MTU */
+	uint32_t mtu;
```

Comment:
The IPv6 PMTU is specified as uint32_t in the RFC. We use only 16 bit value internally, but I thought that it might be good to use the length specified in the RFC, just in case we want to implement this internally too in the future. This way there is no need to change application facing interface.